docs/topics/xml-rule-when-syntax.adoc

```diff
@@ -535,3 +536,89 @@ The `has-classification` element is used to determine if a specified classificat
 |String
 |An optional title to match the classification against.
 |====
+
+[[graph-query-syntax]]
+== <graph-query> Syntax
+
+=== Summary
+
+Use the `<graph-query>` element to search the generated graph for any elements. This element is primarily used to determine if specific classes are embedded within an archive, such as a JAR or EAR. For a better understanding of the `<graph-query>` condition, see the JavaDoc for the http://windup.github.io/windup/docs/latest/javadoc/org/jboss/windup/config/condition/GraphCondition.html[GraphCondition] class. 
+
+The following is an example of a rule that tests to determine if any `ehcache` packages are found within the archive.
+
+[source,xml,options="nowrap"]
+----
+<rule id="embedded-cache-libraries-01000">
+    <when>
+        <graph-query discriminator="JarArchiveModel">
+            <property name="fileName" searchType="regex">.*ehcache.*\.jar$</property>
+        </graph-query>
+    </when>
+    <perform>
+        <classification title="Caching - Ehcache embedded library" category-id="cloud-mandatory" effort="5">
+            <description>
+                The application embeds an Ehcache library.
+
+                Cloud readiness issue as potential state information that is not persisted to a backing service.
+            </description>
+        </classification>
+        <technology-tag level="INFORMATIONAL">Ehcache (embedded)</technology-tag>
+    </perform>
+</rule>
+----
+
+=== Construct a <graph-query>
+
+==== <graph-query> Element Attributes
+
+[cols="1,1,3", options="header"]
+|====
+|Attribute Name
+|Type
+|Description
+
+|discriminator
+|CLASS_NAME
```

Comment:
It actually isn't the Class name, it is the value given to the @TypeValue annotation.